spec/license_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'spec_helper'
+require 'rakelib/default_plugins'
+
+describe "Plugin licenses" do
+
+  let(:common_licenses) {
+    Regexp.union([ /mit/,
+                   /apache*/,
+                   /bsd/,
+                   /ruby/])
+  }
+
+  shared_examples "a license test" do
+
+    subject(:gem_name) do |example|
+      example.metadata[:example_group][:parent_example_group][:description]
+    end
+
+    let(:spec) { Gem::Specification.find_all_by_name(gem_name)[0] }
```

Comment:
```find_by_name``` does not work for the case of logstash-core.gemspec.

Test:
```
jruby-1.7.19 :003 > Gem::Specification.find_by_name("logstash-codec-json")
 => #<Gem::Specification:0x447a logstash-codec-json-0.1.7> 
jruby-1.7.19 :004 > Gem::Specification.find_by_name("logstash-core")
 => nil 
jruby-1.7.19 :005 > Gem::Specification.find_all_by_name("logstash-core")
 => [#<Gem::Specification:0x447c logstash-core-2.0.0.dev-java>] 
```

So if I want to check for the main project I should use this ugly notation.